doc/DEVELOPMENT.rst

```diff
@@ -0,0 +1,266 @@
+notably:
+
+- ``OES_depth_texture`` or ``GL_ANGLE_depth_texture``
+- ``OES_standard_derivatives``
+- ``OES_vertex_array_object``
+- ``EXT_frag_depth``
+- ``EXT_instanced_arrays`` or ``ANGLE_instanced_arrays`` or
+  ``NV_instanced_arrays``
+
+For Windows and macOS, you will need Google's ANGLE library for both ES 3.0 and
+2.0. You'll need to check out
+`ANGLE <https://github.com/google/angle>`__ and build it first. Refer to their
+documentation on how to do that.
+
+Once you've compiled ANGLE, enable it with:
+
+::
+
+    meson -Dinstall_angle=true -Dangle_libegl=/path/to/libEGL.{dll,dylib}
+    -Dangle_libgles=/path/to/libGLESv2.{dll,dylib}
+
+Ensure you use the correct file extension for your platform. (``.dll`` for
+Windows, ``.dylib`` for macOS.)
+
+It'll install automatically with ``ninja install`` (as mentioned above).
+
+Development
+-----------
+
+Build Options
+^^^^^^^^^^^^^
+
+You can enable debugging options/output for development purposes:
+
+::
+
+    meson configure -Dbuildtype=debug -Db_ndebug=false
+
+
+This option also helps for speeding up build times:
+
+::
+
+    meson configure -Db_lto=false
+
+
+For debugging actual gameplay, you can set this option and it will enable cheats
+and other 'fast-forward' options by the pressing keys defined in
+``src/config.h``.
+
+::
+
+    meson configure -Ddeveloper=true
+
+
+Coding Style
+^^^^^^^^^^^^
+
+In the ``*.c`` files, tabs are used. In the ``meson.build`` and ``*.py`` files, spaces
+are used.
+
+To help you abide by this standard, you should install
+`EditorConfig <https://github.com/editorconfig>`__ for your preferred editor of
+choice, and load in the file found at ``.editorconfig`` in the root of the
+project.
+
+
+Platform-Specific Tips
+^^^^^^^^^^^^^^^^^^^^^^
+
+Windows
+"""""""
+
+TODO: complete this section.
+
+Linux (Debian/Ubuntu)
+"""""""""""""""""""""
+
+On an apt-based system (Debian/Ubuntu), ensure you have build dependencies
+installed:
+
+::
+
+    apt-get install meson cmake build-essential libsdl2-dev libsdl2-mixer-dev libogg-dev libopusfile-dev libpng-dev libzip-dev libx11-dev libwayland-dev
+
+
+macOS
+"""""
+
+Manual
+~~~~~
+
+On macOS, you need to begin with installing the Xcode Command Line Tools:
+
+::
+
+    xcode-select --install
+
+There are additional command line tools that you'll need. You can acquire those
+by using `Homebrew <https://brew.sh/>`__.
+
+Follow the instructions for installing Homebrew, and then install the following
+tools:
+
+::
+
+    brew install meson cmake pkg-config docutils imagemagick pygments
+
+
+The following dependencies are technically optional, and can be pulled in at
+build-time, but you're better off installing them yourself to reduce compile
+times:
+
+::
+
+    brew install freetype2 libzip opusfile libvorbis webp sdl2
+
+
+You'll also want to set the following environment variables on your shell's
+.zshrc or .bash_profile:
+
+::
+
+   export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig"
```

Comment:
Does it fail to build without this, or just doesn't pick up the openssl dependency? Note that the openssl dependency is *very* optional: currently it's only used for the SHA256 function, which is used by the shader cache. The shader cache is only used by the GLES backends right now (technically, anything that requires the shader transpiler).

But Taisei also has a built-in (possibly slower) SHA256 implementation, used as fallback if openssl is not available. This fallback can also be forced with `-Duse_libcrypto=false`. It really only makes sense to link against openssl for Linux distribution packages, where it's a core component of the system that's pretty much guaranteed to be present.

Reply:
Hmm, okay. It does _work_ without the dependency, but I figured if anyone was building it, it'd be nice to have.

But yeah, it does introduce some overhead. I can get rid of it.